lib/index.js

```diff
@@ -72,6 +72,43 @@ function json2md(data, prefix, _type) {
     }
 }
 
+/**
+ * @param {Array|Object|String} data The input JSON data.
+ * @param {String} prefix A snippet to add before each line.
+ * @return {Promise.<String, Error>} The generated markdown result.
+ */
+json2md.async = (data, prefix, _type) => Promise.resolve().then(() => {
+    prefix = prefix || ""
+    if (typeof data === "string" || typeof data === "number") {
+        return indento(data, 1, prefix)
+    }
+
+    let content = []
+
+    // Handle arrays
+    if (Array.isArray(data)) {
+        const promises = data.map((d, index) => Promise.resolve()
+            .then(() => json2md.async(d, "", _type))
+            .then((result) => indento(result, 1, prefix))
+            .then((result) => {
+                content[index] = result;
+            })
+        )
+        return Promise.all(promises)
+            .then(() => content.join("\n"))
+    } else {
+        let type = Object.keys(data)[0]
+          , func = converters[_type || type]
+
+        if (typeof func === "function") {
+            return Promise.resolve()
+                .then(() => func(_type ? data : data[type], json2md))
+                .then((result) => indento(result, 1, prefix) + "\n")
```

Comment:
Sorry if I'm missing anything, but isn't this still `sync`?

Reply:
The `func` can be either asynchronous or synchronous function.

For examples:

```js
json2md.converters.md = async (filepath) => {
    return await Promise.fromCallback((cb) => FS.readFile(filepath, {encoding: 'utf8'}, cb));
};

json2md.converters.asyncResult = () => Promise.resolve('async');
```